### ISSUE TITLE
Bridge=Yes does not render unless associated with rendered highways/railways

### ISSUE THREAD

**Issue by Rovastar** [gravitystorm/openstreetmap-carto#612]
A bridge by itself does not render in needs a rendered way for it to appear.

Related to #542

**Comment by Jonobennett** [comment]
Another UK example: https://www.openstreetmap.org/way/37548602 - while people do walk along the top of this, I don't think there's a legal right of way.

**Comment by kocio-pl** [comment]
Does osm-carto code contributors plan to deploy this rendering? 

Taginfo says we have now 2 341 965 objects tagged "bridge=yes" (even the second kv pair, "bridge=viaduct", is notable - 41 583 objects), so the evidence that this feature is important is strong:

http://taginfo.openstreetmap.org/tags/bridge=yes

**Comment by dieterdreist** [comment]
2015-02-11 14:36 GMT+01:00 kocio-pl <notifications@github.com>:

> Taginfo says we have now 2 341 965 objects tagged "bridge=yes" (even the
> second kv pair, "bridge=viaduct", is notable - 41 583 objects), so the
> evidence that this feature is important is strong:
>
> http://taginfo.openstreetmap.org/tags/bridge=yes
>

if you read the contributions in this issue you'll find that "bridge=yes"
is not a bridge by itself, it is an attribute that is stating that
something else (e.g. a highway) is _on_ a bridge, i.e. this is only an
implicit way to map a bridge.

**Comment by kocio-pl** [comment]
Sure - I just realized it later... =} 

It means we don't have clear numbers, but still some of the bridges have multiple highways on it (those are probably important by definition and I guess not all of them have the outline tagged) and even those with just one lane can be better shown at maximum zoom level when rendered with the outline.

**Comment by daganzdaanda** [comment]
I think this issue should be closed, since it's misleading. 
A bridge without a way should be tagged as a closed way with man_made=bridge, or as a bridge relation. Issue #436 deals with how it could look.

**Comment by systemed** [comment]
@daganzdaanda No, this issue is for bridges without *rendered* ways, not bridges without ways of any sort. For example, this modest structure: http://weirdnj.com/stories/paulinskill-viaduct/

**Comment by daganzdaanda** [comment]
OK, thanks, I got confused! 
Yes, that nice bridge is correct as a abandoned railway with bridge=yes. But I'd still add the outlines with man_made=bridge, to be complete. Of course, neither is rendered yet...

**Comment by RussNelson** [comment]
So, have we decided that not rendering railway=abandoned is a mistake YET? I mean, c'mon, not rendering the Paulinskill Viaduct? It dominates the landscape in Hainsburg. There is no place you can't see it .... except if you're looking at OpenStreetMap.

I'm sympathetic to the idea that if you can't see it, it shouldn't be rendered. Can I get a little sympathy for the idea that if you CAN see it, it SHOULD be rendered?

I'm also sympathetic to the idea that rendering every railway=abandoned is wrong, because there are places where even *I* cannot see that a railway used to go. So how about this suggestion: we render railway=railbed, for those places where an ordinary mapper can see a railbed, whether on aerials or on site. That way, those of us who can see the Paulinskill Viaduct can tag it as railway=railbed, and we can close this issue once and for all.

**Comment by dieterdreist** [comment]
2015-02-17 10:57 GMT+01:00 kocio-pl <notifications@github.com>:

> It means we don't have clear numbers, but still some of the bridges have
> multiple highways on it (those are probably important by definition and I
> guess not all of them have the outline tagged) and even those with just one
> lane can be better shown at maximum zoom level when rendered with the
> outline.



yes, I am absolutely in favor of tagging/mapping bridges as their own
objects, you also need this for other stuff like wikipedia links, bridge
names (esp. when the street over it has a different name), bridge refs, etc.
The "solutions" to this problem are just silly (bridge_name etc.) if you
can have a simple and logical solution: a bridge object and "name".
Basically these are workarounds that evolved around the common
misunderstanding that bridge=yes would be the way for mapping a bridge
itself and not just an attribute to say: way on a bridge.

**Comment by daganzdaanda** [comment]
> We're not rendering the Paulinskill Viaduct? 
https://www.openstreetmap.org/way/95507414
It's actually a railway=proposed since Feb 2014, probably the time that railway=abandoned stopped being shown. 
I just added an outline with man_made=bridge and selected info from the wikipedia article: 
https://www.openstreetmap.org/way/328801035

I'll change the railway on the bridge to "railway=railbed" since I think this is a good tag and  I don't believe that there is really a current proposal for a new rail on that route. I'll not touch the rest of the railway at the moment, since I don't know where exactly "abandoned" or "railbed" would be best.

**Comment by SomeoneElseOSM** [comment]
What - we're now proposing tagging changes from OSM carto github issues now?

http://taginfo.openstreetmap.org/search?q=railway%3Drailbed

(currently 0 uses)

Well, I suppose it makes a change from telling the style developers how to render stuff from Every Other OSM List :)

**Comment by daganzdaanda** [comment]
Oops ;-) 
I thought that Russ had a proposal for his railway=railbed idea a few months ago??
Well... back to "abandoned" for the time being.

**Comment by RussNelson** [comment]
Yes, I did propose it, and while it seems perfectly sensible, and a solution to everyone's objections in every regard, I got no positive responses. Do I need to make this suggestion in the form of a pull request?

Yes, nobody is using railway=railbed now. That's the whole point -- people don't want railway=abandoned to be rendered because it is probably 30-40% MISUSED. Yet rendering encourages mapping, so if railway=railbed starts getting rendered, then people will be rewarded for fixing all the wrong railway=abandoned's.

**Comment by pnorman** [comment]
Please remember what the issue is about. It is about ways with bridge tags with no other tags that are rendered, it is not specific to railways. This issue would not change if there were changes to what railway tags were rendered.

If you believe there are new tags that should be rendered create a new issue for them, do not discuss them here.

**Comment by RussNelson** [comment]
Please remember that this problem was created by ceasing the rendering of abandoned railroads. If we can fix that problem, this problem will be fixed at the same time.

If  you believe there are new tags that should be rendered which will solve this problem, please discuss them here.

**Comment by pnorman** [comment]
Locking until @gravitystorm gets back. This issue has gone entirely off-topic, and no discussion is being done on how a bridge should look if its underlying feature is not rendered.

**Comment by gravitystorm** [comment]
This issue is closed since it has been driven far off topic.
